src/test/scala/edu/ie3/simona/service/weather/WeatherSourceWrapperSpec.scala

```diff
@@ -271,4 +371,46 @@ case object WeatherSourceWrapperSpec {
     }
   }
 
+  def prepareWeightTestData(
+      weatherSeq: Seq[(Double, Double, Double, Double)],
+      weights: Seq[(Double, Double, Double, Double)]
+  ): (Seq[WeatherData], WeatherData, WeightSum) = {
+    val weatherData = weatherSeq.map { case (diff, dir, temp, wVel) =>
+      WeatherData(
+        Quantities.getQuantity(diff, StandardUnits.SOLAR_IRRADIANCE),
+        Quantities.getQuantity(dir, StandardUnits.SOLAR_IRRADIANCE),
+        Quantities.getQuantity(temp, Units.KELVIN),
+        Quantities.getQuantity(wVel, StandardUnits.WIND_VELOCITY)
+      )
+    }
+
+    val weightedWeather =
+      weatherData.zip(weights).foldLeft(EMPTY_WEATHER_DATA) {
+        case (
+              currentSum,
+              (
+                WeatherData(diffRad, dirRad, temp, windVel),
+                (diffWeight, dirWeight, tempWeight, wVelWeight)
+              )
+            ) =>
+          currentSum.copy(
+            diffRad = currentSum.diffRad.add(diffRad.multiply(diffWeight)),
+            dirRad = currentSum.dirRad.add(dirRad.multiply(dirWeight)),
+            temp = currentSum.temp.add(temp.multiply(tempWeight)),
+            windVel = currentSum.windVel.add(windVel.multiply(wVelWeight))
+          )
+      }
+    val weightSum = weights.foldLeft(WeightSum.EMPTY_WEIGHT_SUM) {
+      case (currentSum, currentWeight) =>
+        currentSum.add(
+          currentWeight._1,
+          currentWeight._2,
+          currentWeight._3,
+          currentWeight._1
```

Comment:
Furthermore a test where not all four weights amount to ~1 would be nice

Reply:
I'm not fully sure, if I understand correctly, what your intention is.

When determining the weights in the `WeatherSource`, it is ensured, that the sum of weights approximately amounts to 1.0. So, having a weight sum unequal to one actually is a non-specified case.

https://github.com/ie3-institute/simona/blob/d80c00b32bf2e1cbe534e8995d05af8768080317/src/main/scala/edu/ie3/simona/service/weather/WeatherSource.scala#L219-L221

Reply:
Well I'd like to have a test where the correct assignment of weight to parameter is tested (so that, for example, the temperature weight is not used for wind velocity).

Ok, given that constraint, would it make sense to have a test where all four values differ but are still within given limits?
So maybe a test that results in weights (0.991, 0.996, 1.004, 1.009), thus verifying their correct application. 

I can for sure provide such a test, since you are probably limited in available time.

Reply:
Added such tests now